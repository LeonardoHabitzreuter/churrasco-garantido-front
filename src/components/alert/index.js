import React from 'react'
import { Alert as ReactAlert } from 'react-bootstrap'
import Button from 'components/button'

const Alert = ({ show, style, handleDismiss, messages }) => (
  show &&
  <ReactAlert bsStyle={style} onDismiss={handleDismiss}>
    {messages.map((message, index) => <p key={index}>{message}</p>)}
    <Button onClick={handleDismiss}>Fechar</Button>
  </ReactAlert>
)

Alert.defaultProps = {
  messages: []
}

export default Alert
