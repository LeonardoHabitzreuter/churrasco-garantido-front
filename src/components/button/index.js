import React from 'react'
import { Button as AntButton } from 'antd'
import styles from './button.styl'
import classnames from 'classnames'

const Button = ({ children, className, bsStyle, type, ...rest }) => (
  <AntButton className={classnames(styles.fontSans, className)} type={bsStyle} htmlType={type} {...rest}>{children}</AntButton>
)

Button.defaultProps = {
  bsStyle: 'primary',
  type: 'submit'
}

export default Button
