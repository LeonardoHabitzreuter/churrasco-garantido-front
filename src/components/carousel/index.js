import React from 'react'
import { Carousel as AntdCarousel } from 'antd'
import PropTypes from 'prop-types'

const Carousel = ({ children, autoplay, dots, effect }) => (
  <AntdCarousel autoplay={autoplay} dots={dots} effect={effect}>
    {children}
  </AntdCarousel>
)

Carousel.defaultProps = {
  effect: 'fade',
  autoplay: true,
  dots: false
}

Carousel.propTypes = {
  effect: PropTypes.string,
  autoplay: PropTypes.bool,
  dots: PropTypes.bool
}

export default Carousel
