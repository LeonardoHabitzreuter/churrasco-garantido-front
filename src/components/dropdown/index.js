import React from 'react'
import { DropdownButton, MenuItem } from 'react-bootstrap'
import PropTypes from 'prop-types'

const getSelectedItemFromIndex = (items, selectedIndex) => items.find((element, index) => index === selectedIndex)

const Dropdown = ({ id, selectedIndex, items, onChange }) => (
  <DropdownButton id={id} bsStyle='default' title={getSelectedItemFromIndex(items, selectedIndex) || ''}>{
    items.map((item, index) => (
      <MenuItem key={index} eventKey={index} active={index === selectedIndex} onSelect={() => onChange(index)}>{item}</MenuItem>
    ))
  }</DropdownButton>
)

Dropdown.propTypes = {
  id: PropTypes.string.isRequired,
  selectedIndex: PropTypes.number,
  items: PropTypes.array,
  onChange: PropTypes.func
}

Dropdown.defaultProps = {
  items: [],
  selectedIndex: null,
  onChange: () => {}
}

export default Dropdown
