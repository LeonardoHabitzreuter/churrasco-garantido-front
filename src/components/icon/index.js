import React from 'react'
import { Icon as AntIcon } from 'antd'
import PropTypes from 'prop-types'

const Icon = ({ type, className }) => <AntIcon type={type} className={className} />

Icon.propTypes = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string
}

export default Icon
