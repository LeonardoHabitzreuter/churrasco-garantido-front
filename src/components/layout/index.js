import React, { Component } from 'react'
import Menu from 'components/menu'
import Icon from 'components/icon'
import Carousel from 'components/carousel'
import { Layout as AntLayout, Breadcrumb } from 'antd'
import styles from './styles.styl'
const Header = AntLayout.Header
const Content = AntLayout.Content
const Footer = AntLayout.Footer
const Sider = AntLayout.Sider

const breadcrumbsByURL = [{
  url: '/',
  breadcrumbs: [
    'Dashboard'
  ]
}, {
  url: '/registerCompany',
  breadcrumbs: [
    'Cadastrar empresa'
  ]
}, {
  url: '/newOrder',
  breadcrumbs: [
    'Pedidos',
    'Novo pedido'
  ]
}, {
  url: '/myOrders',
  breadcrumbs: [
    'Pedidos',
    'Meus pedidos'
  ]
}, {
  url: '/myAccount',
  breadcrumbs: [
    'Minha conta'
  ]
}, {
  url: '/logout',
  breadcrumbs: []
}]

class Layout extends Component {
  state = {
    collapsed: false
  }

  onCollapse (collapsed) {
    this.setState({ collapsed })
  }

  render () {
    return (
      <AntLayout style={{ minHeight: '100vh' }}>
        <Sider
          collapsible
          collapsed={this.state.collapsed}
          onCollapse={value => this.onCollapse(value)}
        >
          <div className='logo'>
            <a href='/'><Icon type='ant-design' className={styles.logo} /></a>
          </div>
          <Menu />
        </Sider>
        <AntLayout>
          <Header style={{ background: '#fff', padding: '10px' }}>
            <Carousel autoplay dots={false} effect='fade'>
              <h3 className={styles.textCenter}>Cadastre uma empresa</h3>
              <h3 className={styles.textCenter}>Requisite uma ordem de compra</h3>
              <h3 className={styles.textCenter}>Veja suas ordens dentro do seu Dashboard!</h3>
            </Carousel>
          </Header>
          <Content style={{ margin: '0 16px' }}>
            <Breadcrumb style={{ margin: '16px 0' }}>{
              breadcrumbsByURL
                .find(element => element.url === window.location.pathname)
                .breadcrumbs.map(text => <Breadcrumb.Item>{text}</Breadcrumb.Item>)
            }</Breadcrumb>
            <div style={{ padding: 24, background: '#fff', overflow: 'auto' }}>
              {this.props.children}
            </div>
          </Content>
          <Footer style={{ textAlign: 'center' }}>
            Projeto churrasco garantido! Â©2018 Criado por Leonardo Habitzreuter
          </Footer>
        </AntLayout>
      </AntLayout>
    )
  }
}

export default Layout
