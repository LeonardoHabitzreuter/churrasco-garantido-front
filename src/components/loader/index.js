import React from 'react'
import { ClipLoader } from 'react-spinners'
import styles from './loader.styl'
import classnames from 'classnames'

const Loader = ({ className, color, ...rest }) => (
  <div className='sweet-loading'>
    <ClipLoader
      className={
        classnames(
          className,
          styles[color]
        )
      }
      {...rest}
    />
  </div>
)

Loader.defaultProps = {
  color: 'blue',
  loading: true
}

export default Loader
