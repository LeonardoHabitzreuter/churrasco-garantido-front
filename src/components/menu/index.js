import React from 'react'
import Icon from 'components/icon'
import { Menu } from 'antd'
const SubMenu = Menu.SubMenu

const reloadPage = ({ key }) => {
  window.location.href = key
}

export default props => (
  <Menu theme='dark' defaultSelectedKeys={[window.location.pathname]} mode='inline' onClick={reloadPage}>
    <Menu.Item key='/'>
      <Icon type='dashboard' />
      <span>Dashboard</span>
    </Menu.Item>
    <Menu.Item key='/registerCompany'>
      <Icon type='form' />
      <span>Cadastrar empresa</span>
    </Menu.Item>
    <SubMenu
      key='orders'
      title={<span><Icon type='shopping-cart' /><span>Pedidos</span></span>}
    >
      <Menu.Item key='/orders/newOrder'>
        <Icon type='plus' />
        <span>Novo pedido</span>
      </Menu.Item>
      <Menu.Item key='/orders/myOrders'>
        <Icon type='table' />
        <span>Meus pedidos</span>
      </Menu.Item>
    </SubMenu>
    <Menu.Item key='/myAccount'>
      <Icon type='user' />
      <span>Minha conta</span>
    </Menu.Item>
    <Menu.Item key='/logout'>
      <Icon type='logout' />
      <span>Sair</span>
    </Menu.Item>
  </Menu>
)
