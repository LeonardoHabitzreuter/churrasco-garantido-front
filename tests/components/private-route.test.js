import React from 'react'
import { shallow } from 'enzyme'
import { PrivateRoute } from '../../src/components'
import api from '../../src/utils/api'
api.getUser = jest.fn()

describe('Private route component', () => {
  test('should renders properly', () => {
    const wrapper = shallow(<PrivateRoute />)
    expect(wrapper).toMatchSnapshot()
  })
})
