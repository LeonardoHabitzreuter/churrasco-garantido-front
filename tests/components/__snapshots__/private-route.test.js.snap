// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private route component should renders properly 1`] = `
<Route
  render={[Function]}
/>
`;
